rust/arrow/src/compute/kernels/sort.rs

```diff
@@ -686,90 +815,124 @@ pub fn lexsort_to_indices(columns: &[SortColumn]) -> Result<UInt32Array> {
     };
 
     let mut value_indices = (0..row_count).collect::<Vec<usize>>();
-    value_indices.sort_by(lex_comparator);
+    let mut len = value_indices.len();
+
+    if let Some(limit) = limit {
+        len = limit.min(len);
+    }
+    sort_by(&mut value_indices, len, lex_comparator);
 
     Ok(UInt32Array::from(
-        value_indices
-            .into_iter()
-            .map(|i| i as u32)
+        (&value_indices)[0..len]
+            .iter()
+            .map(|i| *i as u32)
             .collect::<Vec<u32>>(),
     ))
 }
 
+pub fn partial_sort<T, F>(v: &mut [T], limit: usize, mut is_less: F)
+where
+    F: FnMut(&T, &T) -> Ordering,
+{
+    let (before, _mid, _after) = v.select_nth_unstable_by(limit, &mut is_less);
+    before.sort_unstable_by(is_less);
```

Comment:
I am wondering, why can we use unstable sort here, while we use stable sort for the normal sort kernel?
Shouldn't that be an option of the kernel?

Reply:
There are some reasons for using unstable_sort.  
1. `select_nth_unstable_by` is unstable, so using `before.sort_stable_by`cadon't ensure it's stable.
2. `sort_unstable` is faster than sort_stable, refer to [doc](https://doc.rust-lang.org/std/primitive.slice.html#method.sort_unstable)

Reply:
Clear, thanks @sundy-li.

Maybe some docs can be added about this?

I think it probably is not really a problem, but might be surprising by someone using the kernel. And we might want to see whether we could use a (faster) unstable version of the normal sort kernel as well (I think that could be used by DataFusion as well).

What do you think @alamb